plugins/preparer/loader.go

```diff
@@ -5,7 +5,13 @@ import (
 	"path/filepath"
 )
 
+type pluginLoader = func(string) (*Plugin, error)
```

Comment:
I might attack this at one level higher so you can exercise the public API:

```go
var DefaultPluginLoader = NewPreparerPlugin

type pluginLoader = func(string) (*Plugin, error)

func LoadPlugins(preparerPluginsPath string) (preparerPlugins []*Plugin, err error) {
	return loadPlugins(preparerPluginsPath, DefaultPluginLoader)
}
```

Inside `TestLoadPlugins`:

```go
// test case loop

DefaultPluginLoader = func(path string) (*Plugin, error) {
	return &Plugin{}, nil
}
defer func() {
	DefaultPluginLoader = NewPreparerPlugin
}()

gotPreparerPlugins, err := LoadPlugins(tt.path)

// assertions
```